sechub-shared-kernel/src/main/java/com/mercedesbenz/sechub/sharedkernel/security/JwtDecoderConfiguration.java

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.sharedkernel.security;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+
+import com.mercedesbenz.sechub.sharedkernel.Profiles;
+
+@Configuration
+@Profile(Profiles.OAUTH2)
+class JwtDecoderConfiguration {
+
+    @Bean
+    JwtDecoder jwtDecoder(OAuth2Properties oAuth2Properties) {
```

Comment:
Could it be possible that OAuth2Properties is null?
Is this autowired - without the annotation? Should we add `@Autowired` to make it clear that the properties are injected?

Reply:
No
The Configuration is profiled with oauth2
OAuth2Properties must not be null when using oauth2 profile
in this case spring will throw a missing bean exception

You generally don't need @Autowire here
Knowing that you're inside the Bean definition it should be clear (similar to constructor injection)